schema.go

```diff
@@ -71,3 +59,279 @@ func (schema DatabaseSchema) validateOperations(operations ...Operation) bool {
+	RefType    RefType       `json:"refType,omitempty"`
+}
+
+// TypeString returns a string representation of the (native) column type
+func (column *ColumnSchema) String() string {
+	var flags []string
+	var flagStr string
+	var typeStr string
+	if column.Ephemeral {
+		flags = append(flags, "E")
+	}
+	if column.Mutable {
+		flags = append(flags, "M")
+	}
+	if len(flags) > 0 {
+		flagStr = fmt.Sprintf("[%s]", strings.Join(flags, ","))
+	}
+
+	switch column.Type {
+	case TypeInteger, TypeReal, TypeBoolean, TypeString:
+		typeStr = string(column.Type)
+	case TypeUUID:
+		typeStr = fmt.Sprintf("uuid [%s (%s)]", column.TypeObj.Key.RefTable, column.TypeObj.Key.RefType)
+	case TypeEnum:
+		typeStr = fmt.Sprintf("enum (type: %s): %v", column.TypeObj.Key.Type, column.TypeObj.Key.Enum)
+	case TypeMap:
+		typeStr = fmt.Sprintf("[%s]%s", column.TypeObj.Key.Type, column.TypeObj.Value.Type)
+	case TypeSet:
+		var keyStr string
+		if column.TypeObj.Key.Type == TypeUUID {
+			keyStr = fmt.Sprintf(" [%s (%s)]", column.TypeObj.Key.RefTable, column.TypeObj.Key.RefType)
+		} else {
+			keyStr = string(column.TypeObj.Key.Type)
+		}
+		typeStr = fmt.Sprintf("[]%s (min: %d, max: %d)", keyStr, column.TypeObj.Min, column.TypeObj.Max)
+	default:
+		typeStr = fmt.Sprintf("Unknown Type")
+	}
+
+	return fmt.Sprintf(strings.Join([]string{typeStr, flagStr}, " "))
+}
+
+// UnmarshalJSON unmarshalls a json-formatted column
+func (column *ColumnSchema) UnmarshalJSON(data []byte) error {
+	hasKeyObj := false
+
+	// ColumnJSON represents the known json values for a Column
+	type ColumnJSON struct {
+		Name       string          `json:"name"`
+		TypeRawMsg json.RawMessage `json:"type"`
+		Ephemeral  bool            `json:"ephemeral,omitempty"`
+		Mutable    bool            `json:"mutable,omitempty"`
+	}
+	var colJSON ColumnJSON
+
+	// Unmarshall known keys
+	if err := json.Unmarshal(data, &colJSON); err != nil {
+		return fmt.Errorf("Cannot parse column object %s", err)
+	}
+
+	column.Name = colJSON.Name
+	column.Ephemeral = colJSON.Ephemeral
+	column.Mutable = colJSON.Mutable
+
+	// 'type' migth be a string or an object, let's figure it out
+	var typeString string
+	if err := json.Unmarshal(colJSON.TypeRawMsg, &typeString); err == nil {
+		// This was an easy one. Use the string as our 'extended' type
+		column.Type = ExtendedType(typeString)
+		return nil
+	}
+
+	// 'type' is an object
+	column.TypeObj = &ColumnType{
+		Key:   &BaseType{},
+		Value: nil,
+		Max:   1,
+		Min:   1,
+	}
+
+	// 'type' can be an object defined as:
+	// "key": <base-type>                 required
+	// "value": <base-type>               optional
+	// "min": <integer>                   optional
+	// "max": <integer> or "unlimited"    optional
+
+	// ColumnTypeJSON is used to dynamically decode the ColumnType
+	type ColumnTypeJSON struct {
+		KeyRawMsg   *json.RawMessage `json:"key,omitempty"`
+		ValueRawMsg *json.RawMessage `json:"value,omitempty"`
+		Min         int              `json:"min,omitempty"`
```

Comment:
"Min" is never parsed, it is always 1, which comes from L228.

Reply:
Fixed in #30